network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/VariantManagerImpl.java

```diff
@@ -127,6 +128,26 @@ public void cloneVariant(String sourceVariantId, String targetVariantId, boolean
         cloneVariant(sourceVariantId, Collections.singletonList(targetVariantId), mayOverwrite);
     }
 
+    private void cloneVariantWithOptimisticConcurrency(List<VariantInfos> variantsInfos, int sourceVariantNum, String targetVariantId) {
+        List<VariantInfos> localVariantInfos = new ArrayList<>(variantsInfos);
+        int attempts = 0;
+        while (attempts < MAX_RETRY_ATTEMPTS) {
+            int targetVariantNum = VariantUtils.findFirstAvailableVariantNum(localVariantInfos);
```

Comment:
Not important and not sure if better
```suggestion
             List<VariantInfos> currentVariantInfos = attempts == 0 ? variantsInfos : 
            index.getStoreClient().getVariantsInfos(index.getNetworkUuid(), true);
            int targetVariantNum = VariantUtils.findFirstAvailableVariantNum(currentVariantInfos);
```